core/src/cache.rs

```diff
@@ -52,6 +53,23 @@ impl Cache {
     }
 }
 
+// cache volume to root/volume
+impl Cache {
+    fn volume_path(&self) -> PathBuf {
+        self.root.join("volume")
+    }
+
+    pub fn volume(&self) -> Option<Volume> {
+        let path = self.volume_path();
+        Volume::from_file(path)
+    }
+
+    pub fn save_volume(&self, volume: &Volume) {
```

Comment:
Once `Volume` implements `Copy`, you can pass it by value here instead of reference.